test/core-members-group/CMGRedemptionHandler.t.sol

```diff
@@ -0,0 +1,811 @@
+        assertEq(totalLengthAfterRedeem, 1);
+        assertEq(collateralIdsAfterRedeem[0], bobId);
+        assertEq(balancesAfterRedeem[0], 10 * CRC);
+    }
+
+    function testGroupMintCollateralWithMultipleSafeTransfers() public {
+        testTrustAliceAndBob();
+
+        // Have Alice send 340 CRC via safeTransfer
+        vm.startPrank(alice);
+        mockCircles.mockHub().safeTransferFrom(alice, ICoreMembersGroup(cmGroup).mintHandler(), aliceId, 340 * CRC, "");
+        vm.stopPrank();
+
+        // Have Bob send 410 CRC via safeTransfer
+        vm.startPrank(bob);
+        mockCircles.mockHub().safeTransferFrom(bob, ICoreMembersGroup(cmGroup).mintHandler(), bobId, 410 * CRC, "");
+        vm.stopPrank();
+
+        // Verify Alice's CRC was received and converted to group CRC
+        assertEq(mockCircles.mockHub().balanceOf(alice, aliceId), 660 * CRC);
+        assertEq(mockCircles.mockHub().balanceOf(alice, uint256(uint160(cmGroup))), 340 * CRC);
+
+        // Verify Bob's CRC was received and converted to group CRC
+        assertEq(mockCircles.mockHub().balanceOf(bob, bobId), 590 * CRC);
+        assertEq(mockCircles.mockHub().balanceOf(bob, uint256(uint160(cmGroup))), 410 * CRC);
+
+        // Get group's redemption handler address and check active collateral
+        address redemptionHandler = ICoreMembersGroup(cmGroup).redemptionHandler();
+        (uint256[] memory collateralIds, uint256[] memory balances, uint256 totalLength) =
+            ICMGRedemptionHandler(redemptionHandler).getActiveCollateral();
+
+        // Verify multiple collateral is tracked
+        assertEq(collateralIds.length, 2);
+        assertEq(balances.length, 2);
+        assertEq(totalLength, 2);
+
+        // Verify collateral IDs and balances match deposits
+        assertEq(collateralIds[0], aliceId);
+        assertEq(balances[0], 340 * CRC);
+        assertEq(collateralIds[1], bobId);
+        assertEq(balances[1], 410 * CRC);
+    }
+
+    function testDirectRedemptionViaCMGRedemptionOperator() public {
+        testGroupMintCollateralWithGroupMint();
+
+        // Have Alice redeem 75 of her CRC circles (collateral) from the group
+        vm.startPrank(alice);
+        uint256[] memory redemptionIds = new uint256[](1);
+        uint256[] memory redemptionValues = new uint256[](1);
+        redemptionIds[0] = aliceId; // Alice's own CRC id
+        redemptionValues[0] = 75 * CRC;
+
+        mockCircles.redemptionOperator().redeem(cmGroup, redemptionIds, redemptionValues);
+        vm.stopPrank();
+
+        // Verify Alice's balances after redemption
+        // She minted 101 CRC into gCRC, then redeemed 75 CRC back
+        assertEq(mockCircles.mockHub().balanceOf(alice, aliceId), 974 * CRC); // Original - minted + redeemed
+        assertEq(mockCircles.mockHub().balanceOf(alice, uint256(uint160(cmGroup))), 26 * CRC); // Minted - redeemed
+    }
+
+    function testCanListActiveCollateralAfterMinting() public {
```

Comment:
not clear what is tested here, the code just repeats the same as testGroupMintCollateralWithMultipleSafeTransfers